lib/smux.c

```diff
@@ -206,40 +198,6 @@ smux_socket (void)
   freeaddrinfo(res0);
   if (sock < 0)
     zlog_warn ("Can't connect to SNMP agent with SMUX");
-#else
-  sock = socket (AF_INET, SOCK_STREAM, 0);
-  if (sock < 0)
-    {
-      zlog_warn ("Can't make socket for SNMP");
-      return -1;
-    }
-
-  memset (&serv, 0, sizeof (struct sockaddr_in));
-  serv.sin_family = AF_INET;
-#ifdef HAVE_STRUCT_SOCKADDR_IN_SIN_LEN
-  serv.sin_len = sizeof (struct sockaddr_in);
-#endif /* HAVE_STRUCT_SOCKADDR_IN_SIN_LEN */
-
-  sp = getservbyname ("smux", "tcp");
-  if (sp != NULL) 
-    serv.sin_port = sp->s_port;
-  else
-    serv.sin_port = htons (SMUX_PORT_DEFAULT);
-
-  serv.sin_addr.s_addr = htonl (INADDR_LOOPBACK);
-
-  sockopt_reuseaddr (sock);
-  sockopt_reuseport (sock);
-
-  ret = connect (sock, (struct sockaddr *) &serv, sizeof (struct sockaddr_in));
-  if (ret < 0)
-    {
-      close (sock);
-      smux_sock = -1;
-      zlog_warn ("Can't connect to SNMP agent with SMUX");
-      return -1;
-    }
-#endif
```

Comment:
Hi Donald,

Same as for bgp_socket(). It will be safer to verify that generic socket management works both for IPv4 & IPv6 before applying this patch

Olivier